confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeConfCheck.java

```diff
@@ -44,17 +45,13 @@
 
   private Properties specialProperties;
 
-  public void checkConfig() throws RepeatConfigurationException, IOException {
-
-    String propsDir = ConfigNodeDescriptor.getInstance().getPropsDir();
-    if (propsDir == null) {
-      // Skip configuration check when developer mode or test mode
-      return;
-    }
+  private ConfigNodeConfCheck() {
     specialProperties = new Properties();
+  }
 
+  public void checkConfig() throws ConfigurationException, IOException, StartupException {
     File specialPropertiesFile =
-        new File(propsDir + File.separator + ConfigNodeConstant.SPECIAL_CONF_NAME);
+        new File(conf.getSystemDir() + File.separator + ConfigNodeConstant.SPECIAL_CONF_NAME);
```

Comment:
It's ok for placing the special conf file to the system dir. But the system dir needs to be created first.

Reply:
fixed